src/utils.js

```diff
@@ -185,6 +185,7 @@ export const prepareViewerData = (entries) => {
       endTime = endTime < lastTimeOfEntry ? lastTimeOfEntry : endTime;
       return {
         index,
+        connection: entry.connection,
```

Comment:
can we index set as `entry.connection` as first preference ? and fallback to existing index if open source users don't send connection. This way we ensure consistency for uniqueness.